service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -147,8 +152,21 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
         HttpStatus.OK);
   }
 
+  public static void captureRequestMetrics(RunSetRequest request) {
+    recordInputsInRequest(request.getWorkflowInputDefinitions().size());
+    recordOutputsInRequest(request.getWorkflowOutputDefinitions().size());
+  }
+
   public static List<String> validateRequest(RunSetRequest request, CbasApiConfiguration config) {
     List<String> errorList = new ArrayList<>();
+    errorList.addAll(validateRequestRecordIds(request, config));
+    errorList.addAll(validateRequestInputsAndOutputs(request, config));
+    return errorList;
+  }
+
+  public static List<String> validateRequestRecordIds(
```

Comment:
splitting off different functions for testing different parts of the request seemed like a good way of making the unit tests simpler